extensions/sql-database-projects/src/models/project.ts

```diff
@@ -823,6 +823,55 @@ export class Project implements ISqlProject {
 		await this.addToProjFile(sqlCmdVariableEntry);
 	}
 
+	/**
+	 * Adds DatabaseSource property to the project
+	 * @param databaseSource Source of the database to add
+	 */
+	public async addDatabaseSourceToProjFile(databaseSource: string): Promise<void> {
+		await this.addValueToProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	* Removes DatabaseSource property from the project
+	* @param databaseSource Source of the database to remove
+	*/
+	public async removeDatabaseSourceFromProjFile(databaseSource: string): Promise<void> {
+		await this.removeValueFromProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	 * Get array of all database source values, regardless of if value is 'well-know'
+	 * @returns Array of all database source values, regardless of if value is 'well-know'
+	 */
+	public getRawDatabaseSourceValues(): string[] {
+		const databaseSource = this.evaluateProjectPropertyValue(constants.DatabaseSource);
+
+		if (databaseSource !== undefined) {
+			return databaseSource.split(';');
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * Get string containing all well-known database sources, separated by semicolon
+	 * @returns Well-known database sources, separated by semicolon
+	 */
+	public getWellKnownDatabaseSourceString(): string {
+		let databaseSourceString: string = '';
+		for (let databaseSourceValue of new Set(this.getRawDatabaseSourceValues())) {
```

Comment:
If we want to dedupe (which I think is reasonable), then we should do so case-insensitively. Why won't we just run a regular `for` loop and do our own comparison?